whois-update/src/main/java/net/ripe/db/whois/update/handler/validator/common/SourceValidator.java

```diff
@@ -28,7 +29,7 @@ public class SourceValidator implements BusinessRuleValidator {
     private static final ImmutableList<Action> ACTIONS = ImmutableList.of(Action.CREATE, Action.MODIFY);
     private static final ImmutableList<ObjectType> TYPES = ImmutableList.copyOf(ObjectType.values());
 
-    private static final Set<ObjectType> NON_AUTH_OBJECT_TYPES = ImmutableSet.of(AUT_NUM, ROUTE, ROUTE6);
+    private static final Set<ObjectType> NON_AUTH_OBJECT_TYPES = ImmutableSet.of(AS_SET, AUT_NUM, ROUTE, ROUTE6);
```

Comment:
This will allow the user to set `source: RIPE-NONAUTH` on *any* AS-SET object, we must prevent that unless the AUT-NUM is `RIPE-NONAUTH` by adding validation to the Transformer class